ports/stm32/main.c

```diff
@@ -562,6 +562,9 @@ void stm32_main(uint32_t reset_mode) {
     MP_STATE_PORT(pyb_config_main) = MP_OBJ_NULL;
 
     // Run boot.py (or whatever else a board configures at this stage).
+    #ifdef MICROPY_BOARD_RUN_FROZEN_AT_BOOT
```

Comment:
I think this should go just above the comment on line 564, with a new comment for this added code, something like "// Run optional frozen boot code."

Also, I think the config option should be `MICROPY_BOARD_FROZEN_BOOT_FILE`, to indicate that it needs to specify a filename.

Reply:
Maybe a more generic name and take state? `MICROPY_BOARD_BEFORE_BOOT(&state)`
Then the following could be done:
```c
#define MICROPY_BOARD_BEFORE_BOOT board_before_boot

void board_before_boot(boardctrl_state_t *state) {
    pyexec_frozen_module("_boot.py");
}
```

Reply:
It's actually already generic the way it is (before this PR): a board can define `MICROPY_BOARD_RUN_BOOT_PY` to first execute a frozen module of its choice, then run the default `boardctrl_run_boot_py()`.

The aim of this PR is to just make it easy for a board to do the common thing of executing frozen code before the user `boot.py`.

Reply:
True!